src/CodeGen_Xtensa.cpp

```diff
@@ -786,9 +786,6 @@ void CodeGen_Xtensa::visit(const Load *op) {
             native_lanes = bytes_in_vector / 2;
         }
         bool is_aligned_load = (op->alignment.modulus % native_lanes == 0) && (op->alignment.remainder % native_lanes == 0);
-        if (external_buffers.count(op->name) > 0) {
```

Comment:
Ugh, this seems like it was a legit bug. I think ``` if (external_buffers.count(op->name) > 0) ``` should be replaced by ``` if (op->param.defined()) ``` .

Reply:
yikes, ok, lets hold off on this PR -- I'll let you choose how to clean this up, you are much closer to this code than I